### PR TITLE
Add `uv run` command

### PR DESCRIPTION
**Pull request by zanieb**
Adds `uv run` which executes a command in your current virtual environment.

This is a simple first milestone, lots of remaining work and behavior. The command is hidden.

### CODE REVIEW

**Comment by konstin** [comment]
Some things i learned i'm gonna dump here:

You can load dlopen libpython and use the c api to load libpython, and run `Py_Main`. This is more efficient and has a better integration than a subprocess, but it requires that uv and python have the same libc, which we can't assume.

Again on unix, you use `execv` to have the current process been taken over by a new one (instead of nested and handling subprocess status). This is apparently not supported on windows, where you always spawn a new process (https://github.com/konstin/poc-monotrail/blob/810f7e1ff43c1f95c96d318691241225352357d1/crates/monotrail/src/monotrail.rs#L759-L794)

File: crates/uv/src/cli.rs [hunk 1]
```diff
@@ -1307,6 +1309,17 @@ pub(crate) struct VenvArgs {
     pub(crate) compat_args: compat::VenvCompatArgs,
 }
 
+#[derive(Args)]
+#[allow(clippy::struct_excessive_bools)]
+pub(crate) struct RunArgs {
+    /// Command
```
[the remark this answers is not in the document]
Comment (zanieb) [reply crates/uv/src/cli.rs:1315]: I'm going to reserve the real documentation for when it's a little more nailed down what the interface is. I think I extend these a bit in the next pull request though.

File: crates/uv/src/commands/run.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    };
+
+    // Spawn and wait for completion
+    // Standard input, output, and error streams are all inherited
+    // TODO(zanieb): Throw a nicer error message if the command is not found
+    let mut handle = process.spawn()?;
+    let status = handle.wait().await?;
+
+    // Exit based on the result of the command
+    // TODO(zanieb): Do we want to exit with the code of the child process? Probably.
```
[the remark this answers is not in the document]
Comment (zanieb) [reply crates/uv/src/commands/run.rs:57]: Yeah agree that's just not built into our exit system so I want to do it separately

**Comment by zanieb** [comment]
Thanks for the notes! Do you see a strong benefit to doing `execv` I feel like we might _want_ to be able to take action after the command? I'll include a note about that somewhere regardless as it does seem kind of nice.

**Comment by konstin** [comment]
Not really, `execv` just seemed more elegant

**Pull request merged**